src/go/pt-k8s-debug-collector/dumper/dumper.go

```diff
@@ -415,7 +444,38 @@ func (d *Dumper) getPodSummary(resource, podName, crName string, namespace strin
 		ports = port + ":5432"
 		summCmdName = "sh"
 		summCmdArgs = []string{"-c", "curl https://raw.githubusercontent.com/percona/support-snippets/master/postgresql/pg_gather/gather.sql" +
-			" 2>/dev/null | PGPASSWORD=" + string(pass) + " psql -X --host=127.0.0.1 --port=" + port + " --user=" + user}
+			" 2>/dev/null | PGPASSWORD='" + string(pass) + "' psql -X --host=127.0.0.1 --port=" + port + " --user='" + user + "'"}
+	case "pgv2":
+		var user, pass, port string
+		if d.forwardport != "" {
+			port = d.forwardport
+		} else {
+			port = "5432"
+		}
+		cr, err := d.getCR("perconapgclusters", namespace)
+		if err != nil {
+			return nil, errors.Wrap(err, "get cr")
+		}
+		if cr.Spec.SecretName != "" {
+			user, err = d.getDataFromSecret(cr.Spec.SecretName, "username", namespace)
+		} else {
+			user, err = d.getDataFromSecret(crName+"-pguser-cluster1", "user", namespace)
+		}
+		if err != nil {
+			return nil, errors.Wrap(err, "get user from PostgreSQL users secret")
+		}
+		if cr.Spec.SecretName != "" {
+			pass, err = d.getDataFromSecret(cr.Spec.SecretName, "password", namespace)
+		} else {
+			pass, err = d.getDataFromSecret(crName+"-pguser-cluster1", "password", namespace)
```

Comment:
the secret name depends on the cluster name

Reply:
So this should be something like
```suggestion
			pass, err = d.getDataFromSecret(crName+"-pguser-"+crName, "password", namespace)
```
?

Reply:
yes, but I want to mention that it's not guaranteed to have this user in cluster. if the end user configured `spec.users`, operator doesn't create this user. but we plan to change this logic and have a default user. @nmarukovich am i right?

Reply:
Could you also please confirm if currently this user name is created as:

`<CLUSTER NAME>-<pguser of anything, specified in spec.users>-<CLUSTER NAME>`

So for default configuration this is cluster1-pguser-cluster1 and could be, for example mysupercluster-mysuperuser-mysupercluster? Or other variations are also possible?

Reply:
by default it's `<CLUSTER NAME>-pguser-<CLUSTER NAME>`
if an user is specified in `spec.users` it'll be `<CLUSTER NAME>-pguser-<USERNAME>`
and please note that the secret name is customizable with `spec.users[].secretName`

Reply:
Thanks, implemented! Please review the updated patch.